training/train_instance.py

```diff
@@ -0,0 +1,125 @@
+
+import sys
+import pathlib
+import os
+import subprocess
+import time
+import io
+import yaml
+import signal
+import copy
+
+parent_dir = str(pathlib.Path(__file__).resolve().parents[1])
+autogenerated_config_dir = os.path.join(
+    parent_dir, "train_configs/autogenerated")
+output_files_dir = os.path.join(parent_dir, "output_files")
+
+
+class TrainInstance:
+    def __init__(self, env_path, port):
+        # Public
+        self.__reset()
+        self.env_path = env_path
+        self.port = port
+
+        # Private
+        template_yaml_path = os.path.join(
+            parent_dir, "train_configs/sac_template.yaml")
+        with open(template_yaml_path, 'r') as f:
+            self.template_params = yaml.load(f)
+
+        # Ensure output_files_dir exisits 
+        pathlib.Path(output_files_dir).mkdir(
+            parents=True, exist_ok=True)
+
+        # Ensure autogenerated_config_dir exists
+        pathlib.Path(autogenerated_config_dir).mkdir(
+            parents=True, exist_ok=True)
+
+    def __reset(self):
+        self.inactive = True
+        self.process = None
+        self.id = None
+
+    def train(self, meta_params):
+        """ Launch training instance
+        """
+        self.inactive = False
+        self.meta_params = meta_params
+        self.id = self.__dict_to_string(meta_params)
+        self.__create_config_file()
+        command = self.__get_command()
+        self.output_file = os.path.join(output_files_dir, str(self.id) + ".out")
+        log = open(self.output_file, 'a')
+        self.process = subprocess.Popen(command, stdout=log, stderr=log, shell=True)
+        print("Training started with PID: " + str(self.process.pid))
+
+    def is_done(self):
```

Comment:
Can we really early stop not promising configurations and keep the value anyway? 
- If we early stop training because it converged it's okay because the value we get from `get_val()` is more or less the one we would have got if we let it run until the end
- If we stop training earlier because it is not promising, then the value we get from `get_val()` is not the value we would have got if we let it run until the end. This may damage the Gaussian Process Search, because we are adding a "fake" value.

Reply:
Good point, I was thinking more in not wasting time and try a different spot.
Maybe giving the information "that area is not so good" is not bad as the gp regressor accounts for some noise so even if its not the real value its something.
But you are right, there is a tradeoff here... We can not implement it for now.